### PR TITLE
Fixes #15535 - http ports used by plugins are being logged now.

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

File: lib/proxy/plugin_initializer.rb [hunk 1]
```diff
@@ -290,6 +301,16 @@ def log_used_settings(settings)
     sorted_keys = settings.keys.map(&:to_s).sort.map(&:to_sym) # ruby 1.8.7 doesn't support sorting of symbols
     to_log = sorted_keys.map {|k| "'%s': %s%s" % [k, settings[k], (default_settings.include?(k) && default_settings[k] == settings[k]) ? " (default)" : ""] }.join(", ")
     logger.debug "'%s' settings: %s" % [plugin.plugin_name, to_log]
+    logger.debug("'%s' ports: 'http': %s, 'https': %s" % [plugin.plugin_name,
```
Comment (iNecas) [inline lib/proxy/plugin_initializer.rb]: What's the added value of logging this? We already have the enabled option logged in the logs themselves. Or am I missing something here?

Comment (dmitri-d) [reply lib/proxy/plugin_initializer.rb]: I had people ask me about this. If anything, this is more of a clarification.

File: lib/proxy/plugin_initializer.rb [hunk 2]
```diff
@@ -290,6 +301,16 @@ def log_used_settings(settings)
...
+                                                          ::Proxy::PluginGroup::HTTP_ENABLED.include?(settings[:enabled]),
+                                                          ::Proxy::PluginGroup::HTTPS_ENABLED.include?(settings[:enabled])])
+  end
+
+  def log_provider_settings(settings)
+    default_settings = plugin.plugin_default_settings
+    sorted_keys = settings.keys.map(&:to_s).sort.map(&:to_sym) # ruby 1.8.7 doesn't support sorting of symbols
+    to_log = sorted_keys.map {|k| "'%s': %s%s" % [k, settings[k], (default_settings.include?(k) && default_settings[k] == settings[k]) ? " (default)" : ""] }.join(", ")
```
Comment (iNecas) [inline lib/proxy/plugin_initializer.rb]: It looks like there is a lot of code duplicate between this method and `log_used_settings`, seems like a good candidate to extract the code to some common place

**Comment by dmitri-d** [comment]
- Removed duplication in settings logging code.

File: lib/proxy/plugin_initializer.rb [hunk 3]
```diff
@@ -238,7 +241,15 @@ def start_services(services, container)
 end
 
 module ::Proxy::DefaultSettingsLoader
-  def load_settings(main_plugin_settings = {})
+  def load_plugin_settings
+    load_settings({}, :log_used_settings)
```
Comment (iNecas) [inline lib/proxy/plugin_initializer.rb]: What about naming `log_plugin_settings` vs. `log_user_settings`.

File: lib/proxy/plugin_initializer.rb [hunk 4]
```diff
@@ -254,7 +265,7 @@ def load_settings(main_plugin_settings = {})
 
     plugin.settings = ::Proxy::Settings::Plugin.new({}, settings)
 
-    log_used_settings(settings)
+    send(how_to_log, settings)
```
Comment (iNecas) [inline lib/proxy/plugin_initializer.rb]: Wouldn't `yield if block_given` and mapping the log statements via `block` be better here?

**Comment by iNecas** [comment]
Tested and work well. Last to suggestions inline

File: lib/proxy/plugin_initializer.rb (hunk 3 above)
Comment (dmitri-d) [reply lib/proxy/plugin_initializer.rb]: The need for slightly different logging  reflects one of the differences between plugins and providers -- the latter do not provide public api. Not sure what the rationale behind `log_user_settings` name is...

File: lib/proxy/plugin_initializer.rb (hunk 4 above)
Comment (dmitri-d) [reply lib/proxy/plugin_initializer.rb]: Replaced with a block.

File: lib/proxy/plugin_initializer.rb (hunk 3 above)
Comment (iNecas) [reply lib/proxy/plugin_initializer.rb]: Sry, typo, I was suggesting renaming `log_used_settings` to `log_plugin_settings`

Comment (iNecas) [reply lib/proxy/plugin_initializer.rb]: This would mirror the `load_plugin_settings`/`load_provider_settings` convention

Comment (dmitri-d) [reply lib/proxy/plugin_initializer.rb]: `log_used_settings` is/can be overridden by modules/provides, I'd like to keep the name as is.

**Review by iNecas — APPROVED** [review]

**Comment by iNecas** [comment]
Thanks @witlessbird

**Pull request merged**